src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -841,80 +956,182 @@ else if (block instanceof BlockEndPortal || block instanceof BlockMobSpawner ||
+                    {
+                        if (checkOrRequestItems(stack))
+                        {
+                            return false;
+                        }
+                    }
+                    for (final ItemStack stack : request)
+                    {
+                        if (stack == null)
+                        {
+                            continue;
+                        }
+                        final int slot = worker.findFirstSlotInInventoryWith(stack.getItem(), stack.getItemDamage());
+                        if (slot != -1)
+                        {
+                            new InvWrapper(getInventory()).extractItem(slot, 1, false);
+                            reduceNeededResources(stack);
+                        }
+                    }
+                }
+                
+
+                entity.setUniqueId(UUID.randomUUID());
+                entity.setLocationAndAngles(
+                        entity.posX,
+                        entity.posY,
+                        entity.posZ,
+                        entity.rotationYaw,
+                        entity.rotationPitch);
+                if (!world.spawnEntityInWorld(entity))
+                {
+                    Log.getLogger().info("Failed to spawn entity");
+                }
             }
         }
+        return true;
+    }
+
+    /**
+     * Calculates the working position.
+     * <p>
+     * Takes a min distance from width and length.
+     * <p>
+     * Then finds the floor level at that distance and then check if it does contain two air levels.
+     * @param targetPosition the position to work at.
+     * @return BlockPos position to work from.
+     */
+    @Override
+    public BlockPos getWorkingPosition(final BlockPos targetPosition)
+    {
+        //get length or width either is larger.
+        final int length = currentStructure.getLength();
+        final int width = currentStructure.getWidth();
+        final int distance = Math.max(width, length) + MIN_ADDITIONAL_RANGE_TO_BUILD;
+
+        return getWorkingPosition(distance, targetPosition, 0);
     }
 
     /**
-     * Create a save mining position for the miner.
-     * @param blockToMine block which should be mined or placed.
-     * @return the save position.
+     * Checks if a certain entity is in the world at a certain position already.
+     * @param entity the entity.
+     * @param world the world.
+     * @return true if there.
      */
-    private BlockPos getNodeMiningPosition(BlockPos blockToMine)
+    private static boolean isEntityAtPosition(final Entity entity, final World world)
     {
-        if(((BuildingMiner) getOwnBuilding()).getCurrentLevel() == null || ((BuildingMiner) getOwnBuilding()).getCurrentLevel().getRandomNode() != null)
+        if(world.getEntitiesWithinAABB(entity.getClass(), new AxisAlignedBB(entity.posX, entity.posY, entity.posZ, entity.posX, entity.posY, entity.posZ)).isEmpty())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)